src/coinjoin/client.h

```diff
@@ -93,7 +94,24 @@ class CoinJoinWalletManager {
 
     CCoinJoinClientManager* Get(const std::string& name) const;
 
-    const wallet_name_cjman_map& raw() const { return m_wallet_manager_map; }
+    template <typename Callable>
+    void ForEachCJClientMan(Callable&& func)
+    {
+        LOCK(cs_wallet_manager_map);
+        for (auto&& [_, clientman] : m_wallet_manager_map) {
+            func(clientman);
+        }
```

Comment:
should we use std::for_each here?

Reply:
We would not be able to do the below snippet because the function isn't taking in a pair but one only one half of it
```c++
std::for_each(m_wallet_manager_map.begin(), m_wallet_manager_map.end(), func);
```

Reply:
could do something ```suggestion
        ranges::for_each(m_wallet_manager_map, [&](auto [_, clientman]){func(clientman)});
```

Reply:
@UdjinM6, thoughts?

Reply:
can't use structured bindings there, would be more like 019a780334 instead

Reply:
I don't like to use `std::for_each` if it's not super trivial.

```
ranges::for_each(m_wallet_manager_map, [&](auto [_, clientman]){func(clientman)});
```

it looks like extra more difficult to read, I disagree with this suggestment.

For-each has been a great option when there was no range-loop in C++! But in this particular case it looks difficult to read, difficult to understand. That's a case when 2 lines better than one!